r2dbc/pom.xml

```diff
@@ -0,0 +1,55 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+
+  <parent>
+    <groupId>com.google.cloud.sql</groupId>
+    <artifactId>jdbc-socket-factory-parent</artifactId>
+    <version>1.0.17-SNAPSHOT</version>
+  </parent>
+  <artifactId>r2cbc-connection-factory</artifactId>
+  <packaging>jar</packaging>
+
+  <name>Cloud SQL R2DBC Connection Factory</name>
+  <description>
+    R2DBC ConnectionFactory to connect to a Cloud SQL database without having to deal with IP whitelisting or SSL
+    certificates manually.
+  </description>
+
+  <dependencies>
+    <dependency>
+      <groupId>io.r2dbc</groupId>
+      <artifactId>r2dbc-spi</artifactId>
+      <version>0.8.2.RELEASE</version>
+    </dependency>
+    <dependency>
+      <groupId>io.r2dbc</groupId>
+      <artifactId>r2dbc-postgresql</artifactId>
+      <version>0.8.4.BUILD-SNAPSHOT</version>
```

Comment:
1. Can we use non-snapshot releases for these? 

2. It doesn't make sense to pull these dependencies when the user isn't needing them. We should follow the existing pattern for the other artifacts and use separate artifacts for mysql/postgres.

Reply:
1. The changes that we rely on only exist in snapshot versions currently. We need to wait until the next version is released.

2. Is it ok if I made both mysql and postgresql dependencies optional?

Reply:
I guess my inclination is to separate them for the following reasons:

1. We can scope to `provided`, and the necessary driver will be pulled by maven if the user doesn't specify it
2. It allows for more clear seperation between driver specific logic
3. It's consistent with the existing artifacts

Are there any advantages to using an optional scope instead?

Reply:
I agree with 2 and 3, will separate the module. 

(Not sure what you mean in 1 - AFAIK `provided` dependencies are provided by a runtime such as application server )